src/pykeen/models/resolve.py

```diff
@@ -0,0 +1,195 @@
+        entity_representations=entity_representations,
+        relation_representations=relation_representations,
+    )
+    return model_from_interaction(
+        dimensions=dimensions,
+        interaction=interaction_instance,
+        entity_representations=entity_representations,
+        relation_representations=relation_representations,
+    )
+
+
+def model_from_interaction(
+    dimensions: Mapping[str, Any],
+    interaction: Interaction,
+    entity_representations: EmbeddingSpecificationHint = None,
+    relation_representations: EmbeddingSpecificationHint = None,
+) -> Type[ERModel]:
+    """Build a model class from an interaction class instance."""
+    entity_representations, relation_representations = _normalize_entity_representations(
+        dimensions=dimensions,
+        interaction=interaction.__class__,
+        entity_representations=entity_representations,
+        relation_representations=relation_representations,
+    )
+
+    class ChildERModel(ERModel):
+        def __init__(self, **kwargs) -> None:
+            """Initialize the model."""
+            super().__init__(
+                interaction=interaction,
+                entity_representations=entity_representations,
+                relation_representations=relation_representations,
+                **kwargs,
+            )
+
+    ChildERModel._interaction = interaction
+
+    return ChildERModel
+
+
+def _normalize_entity_representations(
+    dimensions: Mapping[str, int],
+    interaction: Type[Interaction],
+    entity_representations: EmbeddingSpecificationHint,
+    relation_representations: EmbeddingSpecificationHint,
+) -> Tuple[
+    Sequence[Union[EmbeddingSpecification, RepresentationModule]],
+    Sequence[Union[EmbeddingSpecification, RepresentationModule]],
+]:
+    if entity_representations is None:
+        # TODO: Does not work for interactions with separate tail_entity_shape (i.e., ConvE)
+        if interaction.tail_entity_shape is not None:
+            raise NotImplementedError
+        entity_representations = [
+            EmbeddingSpecification(shape=tuple(
+                dimensions[d]
+                for d in shape
+            ))
+            for shape in interaction.entity_shape
+        ]
+    elif not isinstance(entity_representations, Sequence):
+        entity_representations = [entity_representations]
+    if relation_representations is None:
+        relation_representations = [
+            EmbeddingSpecification(shape=tuple(
+                dimensions[d]
+                for d in shape
+            ))
+            for shape in interaction.relation_shape
+        ]
+    elif not isinstance(relation_representations, Sequence):
+        relation_representations = [relation_representations]
+    return entity_representations, relation_representations
+
+
+# FIXME this function is borrowed from PR #107 and originally implemented by @mberr.
+#  It needs a bit of work
+def _resolve_representations(
```

Comment:
@mberr `_resolve_representations()` isn't used... should we delete it?

Reply:
Yes, I think we can continue with the simpler `_normalize_entity_representations` for now.